velox/common/memory/Memory.cpp

```diff
@@ -63,13 +63,20 @@ uint64_t MemoryPool::getChildCount() const {
 }
 
 void MemoryPool::visitChildren(std::function<void(MemoryPool*)> visitor) const {
-  folly::SharedMutex::ReadHolder guard{childrenMutex_};
-  for (const auto& entry : children_) {
-    auto childPtr = entry.second.lock();
-    if (childPtr != nullptr) {
-      visitor(childPtr.get());
+  std::vector<std::shared_ptr<MemoryPool>> children;
```

Comment:
nit: perhaps, add a comment to explain why we have to call visitor ouside of the lock.